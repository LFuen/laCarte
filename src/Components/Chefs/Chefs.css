.chefPage {
    margin-top: 25px;
    margin-bottom: 100px;
}

.contain{
    display: flex;
    align-content: center;
    justify-content: center;
}

.box {
    float: left;
    width: 33.33%;
    padding: 50px;
}

.chefImageContainer {
    margin-top:50px;
}

.chefList {
    font-size: 20px;
    font-family: 'Sacramento', cursive;
    font-family: 'Special Elite', cursive;
    text-decoration: none;
    color: rgb(0, 0, 0);
}

.h1Chefs {
    font-size: 60px;
    font-weight: bold;
    text-decoration: underline;
}

