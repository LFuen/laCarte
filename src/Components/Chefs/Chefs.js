import React, {Component} from 'react'
import LaCarteContext from '../../context/LaCarteContext'
import {Link} from 'react-router-dom'
import './Chefs.css'


class Chefs extends Component {

    static contextType = LaCarteContext;

    render() {
        const {chefs} = this.context
        return (
                <div className='chefPage'>
                    <h1 className='h1Chefs'>CHEFS</h1>
                    <div className='chefList'>
                        <section className='contain'>
                            <div className='chefImageContainer'>
                                <span title='Chef!'>
                                <img alt='chefMoji' src='https://lfuen.github.io/laCarte/images/chefmoji.png'/>
                                </span>
                            </div>
                            <div className='box'>  
                            {chefs.map(chef => { return (
                            <Link to={`/chefs/${chef.chef_name}`} key={chef.id}><h4>{chef.chef_name}</h4></Link>
                            )})}
                            </div>
                        </section>
                    </div>
                </div> 
        )
    }
}



export default Chefs