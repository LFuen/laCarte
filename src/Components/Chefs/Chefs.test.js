import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import {BrowserRouter} from 'react-router-dom'
import LaCarteContext from '../../context/LaCarteContext'
import Chefs from './Chefs'



describe(`Chefs Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const chefs = renderer.create(<BrowserRouter><Chefs/></BrowserRouter>)
                expect(chefs).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <Chefs />
                    </LaCarteContext.Provider>                
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})