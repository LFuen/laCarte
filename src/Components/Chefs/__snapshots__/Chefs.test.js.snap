// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chefs Component Snapshot renders the UI as expected 1`] = `
<div
  className="chefPage"
>
  <h1
    className="h1Chefs"
  >
    CHEFS
  </h1>
  <div
    className="chefList"
  >
    <section
      className="contain"
    >
      <div
        className="chefImageContainer"
      >
        <span
          title="Chef!"
        >
          <img
            alt="chefMoji"
            src="https://lfuen.github.io/laCarte/images/chefmoji.png"
          />
        </span>
      </div>
      <div
        className="box"
      />
    </section>
  </div>
</div>
`;
